skpro/regression/gradient_boosting/_ngboost.py

```diff
@@ -0,0 +1,342 @@
+        self.dist = dist
+        self.score = score
+        self.estimator = estimator
+        self.natural_gradient = natural_gradient
+        self.n_estimators = n_estimators
+        self.learning_rate = learning_rate
+        self.minibatch_frac = minibatch_frac
+        self.col_sample = col_sample
+        self.verbose = verbose
+        self.verbose_eval = verbose_eval
+        self.init_params = None
+        self.n_features = None
+        self.base_models = []
+        self.scalings = []
+        self.col_idxs = []
+        self.tol = tol
+        self.random_state = check_random_state(random_state)
+        self.best_val_loss_itr = None
+        self.validation_fraction = validation_fraction
+        self.early_stopping_rounds = early_stopping_rounds
+
+        super().__init__()
+
+    def dist_to_ngboost_instance(self):
+        """
+        Convert string to NGBoost object.
+
+        self.dist the input string for the type of Distribution.
+        It then creates an object of that particular NGBoost Distribution.
+
+        Returns
+        -------
+        NGBoost Distribution object.
+        """
+        from ngboost.distns import Laplace, LogNormal, Normal, Poisson, T
+
+        ngboost_dists = {
+            "Normal": Normal,
+            "Laplace": Laplace,
+            "TDistribution": T,
+            "Poisson": Poisson,
+            "LogNormal": LogNormal,
+        }
+        # default Normal distribution
+        dist_ngboost = Normal
+        if self.dist in ngboost_dists:
+            dist_ngboost = ngboost_dists[self.dist]
+
+        return dist_ngboost
+
+    def _fit(self, X, y):
+        """Fit regressor to training data.
+
+        Writes to self:
+            Sets fitted model attributes ending in "_".
+
+        Parameters
+        ----------
+        X : pandas DataFrame
+            feature instances to fit regressor to
+        y : pandas DataFrame, must be same length as X
+            labels to fit regressor to
+
+        Returns
+        -------
+        self : reference to self
+        """
+        from ngboost import NGBRegressor
+        from ngboost.scores import LogScore
+        from sklearn.tree import DecisionTreeRegressor
+
+        if self.estimator is None:
+            self.estimator = DecisionTreeRegressor(
+                criterion="friedman_mse",
+                min_samples_split=2,
+                min_samples_leaf=1,
+                min_weight_fraction_leaf=0.0,
+                max_depth=3,
+                splitter="best",
+                random_state=None,
+            )
+
+        dist_ngboost = self.dist_to_ngboost_instance()
+
+        # Score argument for NGBRegressor
+        ngboost_score = {
+            "LogScore": LogScore,
+        }
+        score = None
+        if self.score in ngboost_score:
+            score = ngboost_score[self.score]
+
+        self.ngb = NGBRegressor(
+            Dist=dist_ngboost,
+            Score=score,
+            Base=self.estimator,
```

Comment:
the estimtaor needs to be cloned - otherwise it will be mutated, and you violate the `sklearn` extension contract (as the failing test tells you)

Reply:
@ShreeshaM07, I think the failure relates to this comment which you have not addressed?

Reply:
I have addressed the clone issue it is not related to that. It is with the handling of the `estimator` which is `DecisionTreeRegressor` but `init` was `None`.